tests/integration/targets/aci_fabric_pod_external_tep/tasks/main.yml

```diff
@@ -0,0 +1,175 @@
+    - add_external_tep.current.0.fabricExtRoutablePodSubnet.attributes.annotation == "orchestrator:ansible"
+
+# CREATE FABRIC POD PROFILE AGAIN TO TEST IDEMPOTENCE
+- name: Create an external tep again
+  cisco.aci.aci_fabric_pod_external_tep:
+    <<: *aci_info
+    podId: 1
+    pool: 10.66.22.0/24
+    addressCount: 5
+    status: active
+    state: present
+  register: add_external_tep_again
+
+- name: Verify external tep creation idempotence
+  assert:
+    that:
+    - add_external_tep_again is not changed
+    - add_external_tep_again.current.0.fabricExtRoutablePodSubnet.attributes.dn == "uni/controller/setuppol/setupp-1/extrtpodsubnet-[10.66.22.0/24]"
+    - add_external_tep_again.current.0.fabricExtRoutablePodSubnet.attributes.pool == "10.66.22.0/24"
+    - add_external_tep_again.current.0.fabricExtRoutablePodSubnet.attributes.reserveAddressCount == "5"
+    - add_external_tep_again.current.0.fabricExtRoutablePodSubnet.attributes.state == "active"
+    - add_external_tep_again.current.0.fabricExtRoutablePodSubnet.attributes.annotation == "orchestrator:ansible"
+
+# QUERY FABRIC EXTERNAL TEP
+- name: query external tep
+  aci_fabric_pod_external_tep:
+    <<: *aci_info
+    podId: 1
+    pool: 10.66.22.0/24
+    state: query
+  register: query_external_tep
+
+- name: Verify external tep query
+  assert:
+    that:
+    - query_external_tep is not changed
+    - query_external_tep.current.0.fabricExtRoutablePodSubnet.attributes.dn == "uni/controller/setuppol/setupp-1/extrtpodsubnet-[10.66.22.0/24]"
+    - query_external_tep.current.0.fabricExtRoutablePodSubnet.attributes.pool == "10.66.22.0/24"
+    - query_external_tep.current.0.fabricExtRoutablePodSubnet.attributes.reserveAddressCount == "5"
+    - query_external_tep.current.0.fabricExtRoutablePodSubnet.attributes.state == "active"
+
+# QUERY ALL EXTERNAL TEPs
+- name: query all external teps
+  aci_fabric_pod_external_tep:
+    <<: *aci_info
+    state: query
+  register: query_external_tep_all
+
+- name: Verify query idempotence
+  assert:
+    that:
+    - query_external_tep_all is not changed
+
+# VERIFY UNABLE TO DELETE ACTIVE EXTERNAL TEP
+- name: Try to delete active external tep
+  aci_fabric_pod_external_tep:
+    <<: *aci_info
+    podId: 1
+    pool: 10.66.22.0/24
+    state: absent
+  register: active_tep_error
+  ignore_errors: true
+
+- name: Verify unable to delete active external tep
+  ansible.builtin.assert:
+    that:
+    - active_tep_error is not changed
+    - active_tep_error is failed
+    - active_tep_error.msg == "APIC Error 105: Unable to delete Routable TEP Pool.Please set the state = inactive."
+
+# CHANGE EXTERNAL TEP TO INACTIVE
+- name: Change external tep to inactive
+  cisco.aci.aci_fabric_pod_external_tep:
+    <<: *aci_info
+    podId: 1
+    pool: 10.66.22.0/24
+    addressCount: 5
+    status: inactive
+    state: present
+  register: external_tep_inactive
+
+# DELETE EXTERNAL TEP
+- name: Delete inactive external tep
```

Comment:
Please add test with checkmode and assert proposed and previous for the test.

Reply:
Done